src/github.com/matrix-org/go-neb/services/imgur/imgur.go

```diff
@@ -0,0 +1,270 @@
+	Images      []imgurGalleryImage `json:"images"`       // images	Array of Images	An array of all the images in the album (only available when requesting the direct album)
+
+	// ** Unimplemented fields **
+	// account_url	string	The account username or null if it's anonymous.
+	// account_id	integer	The account ID of the account that uploaded it, or null.
+	// privacy	string	The privacy level of the album, you can only view public if not logged in as album owner
+	// layout	string	The view layout of the album.
+	// views	integer	The number of image views
+	// ups	integer	Upvotes for the image
+	// downs	integer	Number of downvotes for the image
+	// points	integer	Upvotes minus downvotes
+	// score	integer	Imgur popularity score
+	// vote	string	The current user's vote on the album. null if not signed in or if the user hasn't voted on it.
+	// favorite	boolean	Indicates if the current user favorited the album. Defaults to false if not signed in.
+	// comment_count	int	Number of comments on the gallery album.
+	// topic_id	integer	Topic ID of the gallery album.
+}
+
+type imgurSearchResponse struct {
+	Data    []json.RawMessage `json:"data"`
+	Success bool              `json:"success"` // Request completed successfully
+	Status  int               `json:"status"`  // HTTP response code
+}
+
+// Service contains the Config fields for the Imgur service.
+//
+// Example request:
+//   {
+//			"client_id": "AIzaSyA4FD39..."
+//			"client_secret": "ASdsaijwdfASD..."
+//   }
+type Service struct {
+	types.DefaultService
+	// The Imgur client ID
+	ClientID string `json:"client_id"`
+	// The API key to use when making HTTP requests to Imgur.
+	ClientSecret string `json:"client_secret"`
+}
+
+// Commands supported:
+//    !imgur some_search_query_without_quotes
+// Responds with a suitable image into the same room as the command.
+func (s *Service) Commands(client *gomatrix.Client) []types.Command {
+	return []types.Command{
+		types.Command{
+			Path: []string{"imgur", "help"},
+			Command: func(roomID, userID string, args []string) (interface{}, error) {
+				return usageMessage(), nil
+			},
+		},
+		types.Command{
+			Path: []string{"imgur"},
+			Command: func(roomID, userID string, args []string) (interface{}, error) {
+				return s.cmdImgurImgSearch(client, roomID, userID, args)
+			},
+		},
+	}
+}
+
+// usageMessage returns a matrix TextMessage representation of the service usage
+func usageMessage() *gomatrix.TextMessage {
+	return &gomatrix.TextMessage{"m.notice",
+		`Usage: !imgur image_search_text`}
+}
+
+func (s *Service) cmdImgurImgSearch(client *gomatrix.Client, roomID, userID string, args []string) (interface{}, error) {
+
+	if len(args) < 1 {
+		return usageMessage(), nil
+	}
+
+	// Get the query text to search for.
+	querySentence := strings.Join(args, " ")
+
+	searchResultImage, searchResultAlbum, err := s.text2imgImgur(querySentence)
```

Comment:
`text2imgImgur` : Redundant names are redundant. I know it's to do with imgur, this is the name of the package. Ditto for this function `cmdImgurImgSearch`.